src/wallet/redux/sagas/blockchain.ts

```diff
@@ -0,0 +1,34 @@
+import * as blockchainActions from '../actions/blockchain';
+import { take, put, actionChannel } from 'redux-saga/effects';
+// @ts-ignore
+import simpleAdjudicatorArtifact from 'fmg-simple-adjudicator/contracts/SimpleAdjudicator.sol';
+import contract from 'truffle-contract';
+import detectNetwork from 'web3-detect-network';
+// The blockchain saga will be responsible for dealing with the blockchain using truffle
+export function* blockchainSaga() {
+  const channel = yield actionChannel(blockchainActions.BLOCKCHAIN_DEPLOYADJUDICATOR);
+  while (true) {
+    const action = yield take(channel);
+    switch (action.type) {
+      case blockchainActions.BLOCKCHAIN_DEPLOYADJUDICATOR:
+        const simpleAdjudicatorContract = contract(simpleAdjudicatorArtifact);
+
+        const network = yield detectNetwork(web3.currentProvider);
+        if (!Object.keys(simpleAdjudicatorContract.networks).find(id => id === network.id)) {
```

Comment:
This is interesting. I'm guessing this means that we're actually shipping an app running on multiple networks?